### PR TITLE
[REVIEW] fea/hash_index

### PR DESCRIPTION
**Pull request by quasiben**
This PR adds method to create a hash of the index.  This is a first step for resolving failures in https://github.com/rapidsai/dask-cudf/pull/262

cc @kkraus14

### CODE REVIEW

File: python/cudf/dataframe/index.py [hunk 1]
```diff
@@ -530,6 +531,31 @@ def find_label_range(self, first, last):
...
+
+        _hash = cpp_hash.hash_columns([self.as_column()],
+                                      result, initial_hash_values)
+
+        sr = Series(_hash)
+
+        # hash_columns produces negative valuesg
+        # probably can switch to np.uint32
+        # when supported by libcud
+        return abs(sr)
```
Comment (kkraus14) [inline python/cudf/dataframe/index.py]: Just to confirm, you want to return a Series here and not a subtype of Index?

Comment (quasiben) [reply python/cudf/dataframe/index.py]: I think so -- i need to build some things in dask to properly check now

File: python/cudf/dataframe/index.py [hunk 2]
```diff
@@ -530,6 +531,31 @@ def find_label_range(self, first, last):
...
+        result = NumericalColumn(data=buf, dtype=buf.dtype)
+
+        _hash = cpp_hash.hash_columns([self.as_column()],
+                                      result, initial_hash_values)
+
+        sr = Series(_hash)
+
+        # hash_columns produces negative valuesg
+        # probably can switch to np.uint32
+        # when supported by libcud
```
Comment (kkraus14) [inline python/cudf/dataframe/index.py]:
```suggestion
        # when supported by libcudf
```

**Comment by kkraus14** [comment]
@quasiben I'm going to hold off on merging this until you give the go ahead that no more changes are needed to support dask.

**Comment by quasiben** [comment]
@kkraus14, thanks.  I think we should hold off as well

**Comment by quasiben** [comment]
I think https://github.com/rapidsai/cudf/issues/1720 may be blocking this PR.

**Comment by kkraus14** [comment]
Going to close this for now and we can reopen in the future if we want/need to.

**Pull request closed without merging**